sklearn/cluster/_bisect_k_means.py

```diff
@@ -0,0 +1,729 @@
+                centers_init,
+                max_iter=self.max_iter,
+                verbose=self.verbose,
+                tol=self.tol,
+                x_squared_norms=x_squared_norms,
+                n_threads=self._n_threads,
+            )
+
+            # allow small tolerance on the inertia to accommodate for
+            # non-deterministic rounding errors due to parallel computation
+            if best_inertia is None or inertia < best_inertia * (1 - 1e-6):
+                best_labels = labels
+                best_centers = centers
+                best_inertia = inertia
+
+        distinct_clusters = len(set(best_labels))
+        if distinct_clusters != 2:
+            warnings.warn(
+                "Number of distinct clusters ({}) found smaller than "
+                "n_clusters ({}). Possibly due to duplicate points "
+                "in X.".format(distinct_clusters, 2),
+                ConvergenceWarning,
+                stacklevel=2,
+            )
+
+        return best_centers, best_labels
+
+    def fit(self, X, y=None, sample_weight=None):
+        """Compute bisecting k-means clustering.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+
+            Training instances to cluster.
+
+            .. note:: The data will be converted to C ordering,
+                which will cause a memory copy
+                if the given data is not C-contiguous.
+
+        y : Ignored
+            Not used, present here for API consistency by convention.
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            The weights for each observation in X. If None, all observations
+            are assigned equal weight.
+
+        Returns
+        -------
+        self
+            Fitted estimator.
+        """
+        X = self._validate_data(
+            X,
+            accept_sparse="csr",
+            dtype=[np.float64, np.float32],
+            order="C",
+            copy=self.copy_x,
+            accept_large_sparse=False,
+        )
+
+        self._check_params(X)
+        random_state = check_random_state(self.random_state)
+        sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
+        self._n_threads = _openmp_effective_n_threads()
+
+        if self.algorithm == "full":
+            self._kmeans_single = _kmeans_single_lloyd
+            self._check_mkl_vcomp(X, X.shape[0])
+        else:
+            self._kmeans_single = _kmeans_single_elkan
+
+        _inertia = _inertia_sparse if sp.issparse(X) else _inertia_dense
+
+        if self.verbose:
+            print("Running Bisecting K-Means with parameters:")
+            print(f"-> number of clusters: {self.n_clusters}")
+            print(f"-> number of centroid initializations: {self.n_init}")
+            print(f"-> relative tolerance: {self.tol:.4e}")
+            print(f"-> bisect strategy: {self.bisect_strategy} \n")
```

Comment:
These are just the raw parameters of the estimator. I don't think there's much value in printing them

Reply:
I've treated mode `verbose`  more as a debuging mode. I might remove that if its unconvienient